rosmon_core/src/monitor/node_monitor.cpp

```diff
@@ -174,11 +175,27 @@ void NodeMonitor::start()
 
 	if(m_launchNode->coredumpsEnabled() && g_coreIsRelative && m_processWorkingDirectory.empty())
 	{
-		char tmpfile[256];
-		strncpy(tmpfile, "/tmp/rosmon-node-XXXXXX", sizeof(tmpfile));
-		tmpfile[sizeof(tmpfile)-1] = 0;
-		m_processWorkingDirectory = mkdtemp(tmpfile);
-		m_processWorkingDirectoryCreated = true;
+		if (const char* logDir = std::getenv("ROSMON_LOG_PATH")) 
+		{
+			std::string dir(logDir);
+			dir = dir + "/rosmon/core_dumps";
+			if (chdir(dir.c_str()) == 0 || mkdir(dir.c_str(), 0777) == 0) 
+			{
+				m_processWorkingDirectory = dir;
+				m_processWorkingDirectoryCreated = true;
+			} 
+			else 
+			{
+				perror("Could not create rosmon/core_dumps directory");
+			}
+		}
+		else 
+		{
+			char tmpfile[256];
+			strncpy(tmpfile, "/tmp/rosmon-node-XXXXXX", sizeof(tmpfile));
+			tmpfile[sizeof(tmpfile)-1] = 0;
+			m_processWorkingDirectory = mkdtemp(tmpfile);
+                        					}
```

Comment:
please check the formatting, as mentioned before rosmon code uses tabs instead of spaces